dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/TimeoutFilter.java

```diff
@@ -36,21 +37,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger(TimeoutFilter.class);
 
+    private static final String TIMEOUT_FILTER_START_TIME = "timeout_filter_start_time";
+
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
-        long start = System.currentTimeMillis();
-        Result result = invoker.invoke(invocation);
-        long elapsed = System.currentTimeMillis() - start;
-        if (invoker.getUrl() != null
-                && elapsed > invoker.getUrl().getMethodParameter(invocation.getMethodName(),
-                "timeout", Integer.MAX_VALUE)) {
-            if (logger.isWarnEnabled()) {
-                logger.warn("invoke time out. method: " + invocation.getMethodName()
-                        + " arguments: " + Arrays.toString(invocation.getArguments()) + " , url is "
-                        + invoker.getUrl() + ", invoke elapsed " + elapsed + " ms.");
+        if (invocation.getAttachments() != null) {
+            long start = System.currentTimeMillis();
+            invocation.getAttachments().put(TIMEOUT_FILTER_START_TIME, String.valueOf(start));

```

Comment:
Be careful that this attachment `TIMEOUT_FILTER_START_TIME ` will be passed throughout the RPC chain because of the drawback of RpcContext. #3017

Reply:
Now I think the previous way you give is better, please see my comments below. 
So what do you think now? Would you please revert the code?

Reply:
I have checked the code.
As you said, the timeoutfilter does not affect the RpcContext, nor does it affect the invocation downstream of the call chain. I have reverted the code.